rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -222,6 +220,72 @@ impl Default for SortOptions {
     }
 }
 
+/// Sort primitive values
+fn sort_boolean(
+    values: &ArrayRef,
+    value_indices: Vec<u32>,
+    null_indices: Vec<u32>,
+    nan_indices: Vec<u32>,
```

Comment:
This parameter and the nan handling below (`nans`, `nans_len`, `partial_cmp` instead of `cmp`) could be removed.